app/models/miq_server/log_management.rb

```diff
@@ -208,7 +208,7 @@ def post_current_logs(taskid, log_depot)
   end
 
   def delete_old_requested_logs
-    LogFile.destroy_all(:historical => false, :resource_id => id, :resource_type => self.class.name)
+    LogFile.destroy_all(:historical => false, :resource => self)
```

Comment:
If this isn't already deprecated, I believe it will be in Rails 5. Use `where(..).destroy_all` instead.